### PR TITLE
Following Users

### PR DESCRIPTION
**Pull request by brokenheart1712**

### CODE REVIEW

File: app/controllers/relationships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  def destroy
+    @user = Relationship.find(params[:id]).followed
+    current_user.unfollow(@user)
+    respond_to do |format|
+      format.html { redirect_to @user }
+      format.js
+    end
+  end
+end
```
Comment (leduytungframgia) [inline app/controllers/relationships_controller.rb]: add new line

Comment (brokenheart1712) [inline app/controllers/relationships_controller.rb]: Added

File: app/controllers/relationships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+  def create
+    @user = User.find params[:followed_id]
+    current_user.follow(@user)
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: remove `()`

File: app/controllers/relationships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    respond_to do |format|
+      format.html { redirect_to @user }
+      format.js
+    end
+  end
+
+  def destroy
+    @user = Relationship.find(params[:id]).followed
+    current_user.unfollow(@user)
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: remove `()`

File: app/controllers/relationships_controller.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+  def create
+    @user = User.find params[:followed_id]
+    current_user.follow(@user)
+    respond_to do |format|
+      format.html { redirect_to @user }
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: coding convention

File: app/controllers/relationships_controller.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      format.js
+    end
+  end
+
+  def destroy
+    @user = Relationship.find(params[:id]).followed
+    current_user.unfollow(@user)
+    respond_to do |format|
+      format.html { redirect_to @user }
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: coding convention

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -1,5 +1,5 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, only: [:edit, :update]
+  before_action :logged_in_user, only: [:index, :edit, :update, :following, :followers]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: using `except`

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -39,6 +39,20 @@ def update
     end
   end
 
+  def following
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: Restful

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -39,6 +39,20 @@ def update
...
+    @title = t :following
+    @user  = User.find params[:id]
+    @users = @user.following.paginate page: params[:page]
+    render 'show_follow'
+  end
+
+  def followers
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: Restful

File: app/views/users/_follow.html.erb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<%= form_for(current_user.active_relationships.build, remote: true) do |f| %>
+<div>
```
Comment (dieunb) [inline app/views/users/_follow.html.erb]: indent

File: app/views/users/_follow.html.erb [hunk 10]
```diff
@@ -0,0 +1,6 @@
...
+  <%= hidden_field_tag :followed_id, @user.id %>
+</div>
+<%= f.submit t(:follow), class: "btn btn-primary" %>
```
Comment (dieunb) [inline app/views/users/_follow.html.erb]: indent

**Pull request merged**